Makefile

```diff
@@ -31,9 +31,13 @@ clean:
 distclean: clean
 	find vendor/ -maxdepth 1 -mindepth 1 -not -name vendor.json -exec rm {} -rf \;
 	rm -f manifest/*.yaml
+	rm -f .glide.status
 
-sync:
-	glide install
+sync: .glide.status
+	test -d vendor && \
+	test -f .glide.status && \
+	test '!' .glide.status -ot vendor || \
+		glide install --strip-vendor && touch .glide.status
```

Comment:
Yes, so why not doing something like:

```
build: .glide.status fmt vet compile

.glide.status: glide.yaml
  glide update --strip-vendor && touch $@

sync:
  glide install --strip-vendor
```

Then you have a lightweight check when you build code, which should give you a reasonable good dependency update experience, and a `make sync` target, which allows you to be sure that everything is up to date.